src/storage-client/src/source/persist_source.rs

```diff
@@ -20,22 +21,34 @@ use timely::communication::Push;
 use timely::dataflow::channels::pact::Pipeline;
 use timely::dataflow::channels::Bundle;
 use timely::dataflow::operators::generic::builder_rc::OperatorBuilder;
-use timely::dataflow::operators::{Capability, OkErr};
+use timely::dataflow::operators::{Capability, Operator};
 use timely::dataflow::{Scope, Stream};
 use timely::progress::Antichain;
 use timely::scheduling::Activator;
 
 use mz_expr::MfpPlan;
 use mz_persist_client::cache::PersistClientCache;
 use mz_persist_client::fetch::FetchedPart;
-use mz_repr::{DatumVec, Diff, GlobalId, Row, Timestamp};
+use mz_repr::{Diff, GlobalId, Row, Timestamp};
 
 use crate::controller::CollectionMetadata;
 use crate::types::errors::DataflowError;
 use crate::types::sources::SourceData;
 
 pub use mz_persist_client::operators::shard_source::FlowControl;
 use mz_timely_util::buffer::ConsolidateBuffer;
+use mz_timely_util::operator::CollectionExt;
+
+/// Request a particular guarantee about consolidation from the source.
+/// It is legal for the source to satisfy a request for consolidation with an ordinary consolidate
+/// operator, but the hope is for it to be able to do something more clever.
+pub enum Consolidation {
+    /// No guarantees about how the resulting collection's data is consolidated.
+    Maybe,
+    /// Guarantees that the resulting collections is consolidated, but not necessarily exchanged
+    /// the way the normal consolidate operator would do it.
```

Comment:
I don't think this is true. Only the inputs to the MFP would be consolidated, but its output wouldn't be.